Part of the fix for upstream issue 5078.  This also avoids the crash
when logging into the native mga display on sparc servers.

https://gitlab.gnome.org/GNOME/mutter/-/issues/2267
https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5078

--- a/src/x11/session.c	2022-05-31 12:16:35.771345666 -0700
+++ b/src/x11/session.c	2022-05-31 12:18:06.754018419 -0700
@@ -113,7 +113,8 @@
        * being cleaned up, since it is owned by libSM.
        */
       disconnect ();
-      meta_context_terminate (ice_connection->context);
+      if (ice_connection->context != NULL)
+      	meta_context_terminate (ice_connection->context);
 
       return FALSE;
     }
